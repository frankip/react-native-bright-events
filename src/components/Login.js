import React from 'react'

import { Link } from 'react-router-dom';

import { useFormik } from "formik"

const Login = () => {
    const formik = useFormik({
        initialValues: {
            email: "",
            password: "",
          },
          onSubmit(values) {
            console.log(values)
          }
    })
    return (
        <div className="body">
              <div className="intro">
                <div>
                  <h1>
                    Welcome to<br />
                    <span>Bright Events</span>
                  </h1>
                  <p>
                    create and manage different types of events while making them
                    easily accessible to target markets
                  </p>
                </div>
              </div>
              <div className="form">
                <ul className="tab-group">
                  <li className="tab">
                    <Link to="/signup">Signup</Link>
                  </li>
                  <li className="tab active">
                    <Link to="/login">login</Link>
                  </li>
                </ul>
                <div className="tab-content">
                  <div id="login">
                    <h3>Welcome Back!</h3>
                    <form onSubmit={formik.handleSubmit}>
                      <div className="field-wrap">
                        <input 
                        type="email" 
                        name="email" 
                        placeholder="Email" 
                        required 
                        onChange={formik.handleChange}
                        value={formik.values.name} />
                      </div>
                      <div className="field-wrap">
                      <input 
                        type="password" 
                        name="password" 
                        placeholder="password" 
                        required 
                        onChange={formik.handleChange}
                        value={formik.values.name} />
                      </div>
                      <p className="forgot">
                        <Link to="/signup">Forgot Password?</Link>
                      </p>
                      <button value="submit" className="button button-block">
                        Log In
                      </button>
                    </form>
                  </div>
                  <div>
                  </div>
                </div>
              </div>
            </div>
    )
}

export default Login
