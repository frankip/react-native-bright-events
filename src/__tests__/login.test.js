
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import expect from "expect";
import Login from '../components/Login';

describe('Login Component', () => {
  const wrapper = shallow(<Login />);
  // const preventDefault = jest.fn();
  // it('renders without crashing', () => {
  //   expect(shallowToJson(wrapper)).toMatchSnapshot();
  // });
  it('renders state initially', () => {
    expect(wrapper.state().email).toEqual('');
    expect(wrapper.state().password).toEqual('');
  });
  it('renders appropriate number of inputs', () => {
    expect(wrapper.find('input').length).toEqual(2);
  });
  it('shows a submit button', () =>{
    expect(wrapper.find('.button').length).toEqual(1)
  })
});
