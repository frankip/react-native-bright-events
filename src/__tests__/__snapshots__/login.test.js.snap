// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders without crashing 1`] = `
<div
  className="body"
>
  <div
    className="intro"
  >
    <div>
      <h1>
        Welcome to
        <br />
        <span>
          Bright Events
        </span>
      </h1>
      <p>
        create and manage different types of events while making them easily accessible to target markets
      </p>
    </div>
  </div>
  <div
    className="form"
  >
    <ul
      className="tab-group"
    >
      <li
        className="tab"
      >
        <Link
          replace={false}
          to="/signup"
        >
          Signup
        </Link>
      </li>
      <li
        className="tab active"
      >
        <Link
          replace={false}
          to="/login"
        >
          login
        </Link>
      </li>
    </ul>
    <div
      className="tab-content"
    >
      <div
        id="signup"
      >
        <h3>
          Welcome Back!
        </h3>
        <form
          method="POST"
          onSubmit={[Function]}
        >
          <div
            className="field-wrap"
          >
            <label>
               
              Email Address
              <span
                className="req"
              >
                *
              </span>
               
            </label>
            <input
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
            />
          </div>
          <div
            className="field-wrap"
          >
            <label>
               
              Set A Password
              <span
                className="req"
              >
                *
              </span>
               
            </label>
            <input
              name="password"
              onChange={[Function]}
              required={true}
              type="password"
            />
          </div>
          <p
            className="forgot"
          >
            <Link
              replace={false}
              to="/"
            >
              Forgot Password?
            </Link>
          </p>
          <button
            className="button button-block"
            value="submit"
          >
            Log In
          </button>
        </form>
      </div>
      <div
        id="login"
      >
        <h3>
          Welcome Back!
        </h3>
        <form
          action="index.html"
          method="post"
        >
          <div
            className="field-wrap"
          >
            <label>
               
              Email Address
              <span
                className="req"
              >
                *
              </span>
               
            </label>
            <input
              required={true}
              type="email"
            />
          </div>
          <div
            className="field-wrap"
          >
            <label>
               
              Password
              <span
                className="req"
              >
                *
              </span>
               
            </label>
            <input
              required={true}
              type="password"
            />
          </div>
          <p
            className="forgot"
          />
          <a
            className="button button-block"
            href="index.html"
          >
            Log In
          </a>
        </form>
      </div>
    </div>
  </div>
</div>
`;
